Life-Pill/patient-mobile
# Pull Request for mobile app

### Description:
This pull request brings several improvements to the application, with a focus on enhancing the user experience and resolving key issues. The major updates include the implementation of the Bloc architecture and the resolution of an error affecting the splash screen. Additionally, the UI for essential pages, namely the sign-in, sign-up, OTP, and forgot password pages, has been completed.

### Changes Made:

1. Bloc Architecture:
Implemented the Bloc architecture to enhance the application's structure and maintain a clear separation of concerns.

2. Resolved Splash Screen Error:
Identified and fixed an error impacting the splash screen, where the app goes to full screen mode.

3. Sign-In Page:
Completed the UI for the sign-in page, providing users with an intuitive and aesthetically pleasing interface.

4. Sign-Up Page:
Implemented the UI for the sign-up page, making the registration process user-friendly.

5. OTP Page:
Created a visually appealing UI for the OTP page, streamlining the verification process.

6. Forgot Password Page:
Implemented the UI for the forgot password page, ensuring a seamless experience for users recovering their accounts.

### How to Test:
1. Ensure the application is running locally.
2. Navigate through the application to test the functionality of the sign-in, sign-up, OTP, and forgot password pages.
3. Verify that the splash screen loads without going into full screen.

### Additional Notes:
The introduction of the Bloc architecture is aimed at improving maintainability and scalability. Developers are encouraged to review and provide feedback on the implemented structure.
Consideration has been given to the visual design and usability of the UI, but further feedback and adjustments are welcome.

### Checklist:


- [x] Updated documentation to reflect changes related to the Bloc architecture and UI enhancements.

- [x] Code adheres to established style guidelines.

- [x] Commits are accompanied by descriptive messages, providing clarity on the changes made.
